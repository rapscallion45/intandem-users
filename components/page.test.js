import React from 'react';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';
import toJson from 'enzyme-to-json';
import Page from './page';

describe('Page', () => {
  it('Renders correctly enzyme', () => {
    /* Arrange */
    const mockChildren = (
      <div>
        <p>Child 1</p>
        <p>child 2</p>
      </div>
    );

    /* Act */
    const wrapper = shallow(<Page>{mockChildren}</Page>);

    /* Assert */
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('Page Title', () => {
    it('Should render the passed text', () => {
      /* Arrange */
      const mockChildren = (
        <div>
          <p>Child 1</p>
          <p>child 2</p>
        </div>
      );
      const testPageTitle = 'Test Page | Test';

      /* Act */
      const wrapper = render(<Page title={testPageTitle}>{mockChildren}</Page>);

      /* Assert */
      expect(wrapper.getByText(testPageTitle)).toBeInTheDocument();
    });
  });

  describe('Child Components', () => {
    it('Should render passed children components', () => {
      /* Arrange */
      const mockChildren = (
        <div>
          <p>Child 1</p>
          <p>Child 2</p>
        </div>
      );

      /* Act */
      const wrapper = render(<Page>{mockChildren}</Page>);

      /* Assert */
      expect(wrapper.getByText('Child 1')).toBeInTheDocument();
      expect(wrapper.getByText('Child 2')).toBeInTheDocument();
    });
  });
});
