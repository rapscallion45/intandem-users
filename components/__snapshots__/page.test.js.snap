// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Renders correctly enzyme 1`] = `
<ForwardRef(Box)>
  <title />
  <div>
    <p>
      Child 1
    </p>
    <p>
      child 2
    </p>
  </div>
</ForwardRef(Box)>
`;
